src/main/cpp/src/decimal_utils.cu

```diff
@@ -778,10 +798,11 @@ private:
   // output column for overflow detected
   bool * const overflows;
 
+  bool const is_int_div;
```

Comment:
I don't understand `is_int_div` and why that changes the result from 64 bits to 128 bits. Should this be tied to `T`?

Reply:
So there are two types of divide that Spark supports `/` and `DIV`.  The output of `\` is a decimal value where the scale and precision is determined by Spark. The output of `DIV` is a long. `is_int_div` is an attempt to implement `DIV`.  The problem is that to make `DIV` work properly multiple different things need to be set. `T` must be an `int64_t`, scale must be `0`, and `is_int_div` must be `true`. This is all hidden from the end user, so it is mostly okay to me, but if you have suggestions to clean it up that would be great.

The biggest problem that I have is that the kernels have dead paths. The kernel for `DIV` still has `is_int_div` in it which means the paths for both true and false are still compiled and runnable.

Reply:
I have added `static_assert` which should alleviate the problem of instantiating the class with "illegal" types that we don't support.

Reply:
Yeah `static_assert` is good.